Sources/Basic/Lock.swift

```diff
@@ -58,10 +59,31 @@ public final class FileLock {
         self.cachePath = cachePath
     }
 
-    /// Try to aquire a lock. This method will block until lock the already aquired by other process.
+    /// Registers signal handler for SIGUSR1 and raises the signal
+    /// after the given number of seconds.
+    private func setupTimeout(seconds: Double) {
+        // Register signal handler to avoid process termination.
+        var action = sigaction()
+        action.__sigaction_u.__sa_handler = { _ in }
+        sigaction(SIGUSR1, &action, nil)
+        // Raise the signal when we reach timeout.
+        DispatchQueue.global().asyncAfter(deadline: .now() + seconds) {
+            raise(SIGUSR1)
+        }
+    }
+
+    /// Try to aquire a lock. This method will block if a lock is already aquired by other process.
     ///
     /// Note: This method can throw if underlying POSIX methods fail.
-    public func lock() throws {
+    ///
+    /// - Parameters:
+    ///   - timeout: If provided, the method will return in case the lock is not aquired within the timeout value.
+    /// - Returns: True if lock was aquired, false otherwise.
+    @discardableResult
+    public func lock(timeout seconds: Double? = nil) throws -> Bool {
```

Comment:
wasn't sure about semantics to keep there, does it skip the block on timeout? who would check if the lock was actually acquired?